src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/Common.cs

```diff
@@ -13,12 +19,58 @@ static Common()
             TestTracer = new Tracer(settings);
             ServiceName = TestTracer.DefaultServiceName;
 
+            Tracer.Instance = TestTracer;
```

Comment:
Should we really be "resetting" the `Tracer.Instance` like this? What if this module is initialized (for some reason) in non-testing code? We'd overwrite the static tracer instance then, or alternatively overwrite `Tracer.Instance` 🤔

Reply:
Mmm I have to overwrite the tracer instance for integration tests, so new spans are created as children of the test spans. This is required also for billing so those children spans will not be billed by APM. 

But you're right, having this code in a `.cctor` is dangerous, I'll move this to a method and call it before the test span generation, so we ensure we are in a testing scope.

Reply:
done in 61f4f30